app/react/kubernetes/applications/CreateView/application-services/ingress/AppIngressPathForm.tsx

```diff
@@ -0,0 +1,161 @@
+import { RefreshCw, Trash2 } from 'lucide-react';
+import { useEffect, useMemo, useState } from 'react';
+import { UseQueryResult } from 'react-query';
+import { FormikErrors } from 'formik';
+
+import { Ingress } from '@/react/kubernetes/ingresses/types';
+
+import { Select } from '@@/form-components/ReactSelect';
+import { Button } from '@@/buttons';
+import { FormError } from '@@/form-components/FormError';
+import { InputGroup } from '@@/form-components/InputGroup';
+import { Link } from '@@/Link';
+
+import { IngressOption, ServicePortIngressPath } from '../types';
+
+type Props = {
+  ingressPath?: ServicePortIngressPath;
+  ingressPathErrors?: FormikErrors<ServicePortIngressPath>;
+  ingressHostOptions: IngressOption[];
+  onChangeIngressPath: (ingressPath: ServicePortIngressPath) => void;
+  onRemoveIngressPath: () => void;
+  ingressesQuery: UseQueryResult<Ingress[], unknown>;
+  namespace?: string;
+  isEditMode?: boolean;
+};
+
+export function AppIngressPathForm({
+  ingressPath,
+  ingressPathErrors,
+  ingressHostOptions,
+  onChangeIngressPath,
+  onRemoveIngressPath,
+  ingressesQuery,
+  namespace,
+  isEditMode,
+}: Props) {
+  const [selectedIngress, setSelectedIngress] = useState<IngressOption | null>(
+    ingressHostOptions[0]
```

Comment:
is there a chance `ingressHostOptions[0]` is `undefined` or order of `ingressHostOptions` changes?

Reply:
Yeah, there is a possibility of it being undefined, if it is, I'll set it to null (allowed in the select onchange method). There is a possibility of the options changing, which should be handled by the useEffect below